msgpack23: add version 2.1

rwindegger opened the pull request:
### Summary
Changes to recipe:  **msgpack23/2.0**

#### Motivation
Add a new library for MessagePack based on the C++23 standard.

#### Details
Add a new library that supports serialization and deserialization of the MessagePack standard.


---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

rwindegger commented:
@conan-io/barbarians Please check why the pipelines are not running.

rwindegger commented:
@conan-io/barbarians The pipeline is stuck since multiple days is there anything I can do?

ErniGH commented:
Hello @rwindegger ,

First of all, thank you very much for your PR and for your interest in adding your library to Conan.

The main issue with CI is that, in Conan, we usually don’t allow the execution of unit tests in our pipelines. The tests we typically include are focused on verifying that the linker worked correctly and that the symbols were properly linked. In this case, since it is a header-only library, we only need a small test that, by using one of its functionalities, ensures the package has been built correctly. Using GTest and building all unit tests would be excessive.

We would appreciate it if the change could be made directly upstream. CMake provides the `BUILD_TESTING` variable for these cases—you can check this link for an example of how to implement it. https://cmake.org/cmake/help/latest/variable/BUILD_TESTING.html

Additionally, requirements are usually not added to `conandata.yml`. While this approach is sometimes used in recipes with many dependencies, in this case, it is much simpler to add it directly in requirements using `self.requires("gtest/1.15.0")`. 😄

rwindegger commented:
Hello @ErniGH,

thanks for your kind words. I hope I've adjusted upstream in the right way. I've also changed that the dependencies are now directly in the conanfile.py.

Thanks for your support.

rwindegger commented:
Hello @ErniGH,
I'm not really sure why the CI is still failing. It's not finding symbols that are available in C++23, but the standard is set accordingly:
`-- Warning: Standard CMAKE_CXX_STANDARD value defined in conan_toolchain.cmake to 17 has been modified to 23 by /home/conan/workspace/cci_prod_PR-26640/conan-home/p/b/msgpa866cbd09df20c/b/CMakeLists.txt`.
Also it seems the introduced option is not honored, do I need to adjust something else in the conanfile.py?

ErniGH commented:
I attach here the build log in Mac:
[mac-log.txt](https://github.com/user-attachments/files/19344290/mac-log.txt)

ErniGH reviewed (approved):
LGTM Thanks for this new library 😄
